### PR TITLE
TFTRT: Add converter for ArgMin and ArgMax

### PR DESCRIPTION
**Pull request by trevor-m**
Also adds unit tests for ArgMin/ArgMax.

Disables INT32 for all ops using TopK because it is unsupported by TRT.

### CODE REVIEW

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 1]
```diff
@@ -4760,6 +4762,160 @@ TEST_F(OpConverterTest, ConvertPack) {
   // TestConvertPack<DT_INT32>(this);
 }
 
+// Get the NodeDef for ArgMax.
+NodeDef GetArgMaxNodeDef(DataType input_dtype, DataType output_dtype) {
```
Comment (smit-hinsu) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc]: Could we make use template here with types ops::ArgMin or ops::ArgMax?

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 2]
```diff
@@ -4760,6 +4762,160 @@ TEST_F(OpConverterTest, ConvertPack) {
...
+    AddTestWeights<int32>("dimension", {1}, {0});
+    RunValidationAndConversion(
+        node_def, error::UNIMPLEMENTED,
+        "TensorRT does not allow manipulation of the batch dimension, at "
+        "my_argmax");
+  }
+
+  TestConvertArgMinMax<DT_FLOAT>(this);
+  TestConvertArgMinMax<DT_HALF>(this);
+  // TestConvertArgMinMax<DT_INT32>(this);
```
Comment (smit-hinsu) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc]: nit: add a comment explaining the reason.

**Review by smit-hinsu — DISMISSED** [review]
Thanks, Trevor!

**Review by smit-hinsu — DISMISSED** [review]
Looks great!

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 3]
```diff
@@ -4150,13 +4150,62 @@ Status ConvertSoftmax(OpConverterParams* params) {
...
+  std::vector<int> size(dims.d, dims.d + dims.nbDims);
+  size.erase(size.begin() + trt_axis);
+  nvinfer1::Dims new_dims;
+  TF_RETURN_IF_ERROR(TensorShapeArrayToTrtDims(size, &new_dims));
+  nvinfer1::ITensor* output_tensor = nullptr;
+  TF_RETURN_IF_ERROR(params->converter->PrepareTensorForShape(
+      TRT_TensorOrWeights(output_indices_tensor), new_dims,
+      /*validation_only=*/false, &output_tensor));
+
+  params->outputs->push_back(TRT_TensorOrWeights(output_tensor));
```
Comment (smit-hinsu) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: Need to add `return Status::OK();` here.

Comment (trevor-m) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: Thanks

**Review by smit-hinsu — APPROVED** [review]

**Pull request merged**